### ISSUE TITLE
Error while at prototype parsing for pumpjack

### ISSUE THREAD

**Issue by clampi** [kirazy/reskins-bobs#1]
Encountered with reskins-library 0.00.1, reskins-bobs 0.00.2

Happened after enabling both reskin-bob and reskins-library

Relevant error from Factorio log
```log
43.849 Error ModManager.cpp:1508: Error while loading entity prototype "pumpjack" (mining-drill): Animation speed in hr_version doesn't match normal animation speed in property tree at ROOT.mining-drill.pumpjack.animations.north.layers[0].hr_version
```

**Comment by kirazy** [comment]
Could you post your log file?

**Comment by clampi** [comment]
I could, but looking at the log there are no glaring messages standing out from your mod neither anything in particular regarding 'pumpjack' entity, beside the final error. So, I think you're after something else. Right now the log is full of debug messages from some of my own mods, so most is not relevant.

**Comment by kirazy** [comment]
I mostly want your list of mods. In my mod, the animation_speed is set by a single common variable so as-is shouldn't be different. So my current theory is that some other mod is setting pumpjack sprites in some fashion and using the vanilla animation_speed value (I tweak it).

**Comment by kirazy** [comment]
And yes, a serpent print would be useful.

**Comment by clampi** [comment]
I cleaned up the log and isolated just with all the enabled bob mods and your reskin
```log
   0.474 Loading mod settings bobinserters 0.18.3 (settings.lua)
   0.474 Loading mod settings reskins-library 0.0.1 (settings.lua)
   0.474 Loading mod settings bobassembly 0.18.5 (settings.lua)
   0.475 Loading mod settings boblogistics 0.18.9 (settings.lua)
   0.475 Loading mod settings bobmining 0.18.1 (settings.lua)
   0.475 Loading mod settings bobmodules 0.18.3 (settings.lua)
   0.475 Loading mod settings bobpower 0.18.6 (settings.lua)
   0.475 Loading mod settings bobrevamp 0.18.5 (settings.lua)
   0.476 Loading mod settings bobtech 0.18.3 (settings.lua)
   0.476 Loading mod settings bobwarfare 0.18.6 (settings.lua)
   0.476 Loading mod settings bobvehicleequipment 0.18.1 (settings.lua)
   0.476 Loading mod settings reskins-bobs 0.0.3 (settings.lua)
   0.479 Loading mod core 0.0.0 (data.lua)
   0.567 Loading mod base 0.18.30 (data.lua)
   0.841 Loading mod bobinserters 0.18.3 (data.lua)
   0.959 Loading mod boblibrary 0.18.9 (data.lua)
   1.067 Loading mod reskins-library 0.0.1 (data.lua)
   1.173 Loading mod bobassembly 0.18.5 (data.lua)
   1.288 Loading mod bobequipment 0.18.1 (data.lua)
   1.399 Loading mod bobgreenhouse 0.18.1 (data.lua)
   1.510 Loading mod boblogistics 0.18.9 (data.lua)
   1.648 Loading mod bobmining 0.18.1 (data.lua)
   1.767 Loading mod bobmodules 0.18.3 (data.lua)
   1.886 Loading mod bobpower 0.18.6 (data.lua)
   2.025 Loading mod bobrevamp 0.18.5 (data.lua)
   2.146 Loading mod bobtech 0.18.3 (data.lua)
   2.268 Loading mod bobwarfare 0.18.6 (data.lua)
   2.410 Loading mod bobvehicleequipment 0.18.1 (data.lua)
   2.541 Loading mod reskins-bobs 0.0.3 (data.lua)
   2.669 Loading mod base 0.18.30 (data-updates.lua)
   2.797 Loading mod bobinserters 0.18.3 (data-updates.lua)
   2.926 Loading mod bobassembly 0.18.5 (data-updates.lua)
   2.927 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__bobassembly__/prototypes/chemical-plant-updates.lua:78: in main chunk
	[C]: in function 'require'
	__bobassembly__/data-updates.lua:4: in main chunk
   2.927 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology advanced-electronics-3 does not exist.
   3.056 Loading mod bobequipment 0.18.1 (data-updates.lua)
   3.057 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__bobequipment__/prototypes/recipe/updates.lua:57: in main chunk
	[C]: in function 'require'
	__bobequipment__/data-updates.lua:1: in main chunk
   3.057 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology zinc-processing does not exist.
   3.186 Loading mod bobgreenhouse 0.18.1 (data-updates.lua)
   3.315 Loading mod boblogistics 0.18.9 (data-updates.lua)
   3.320 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__boblogistics__/prototypes/technology-updates.lua:159: in main chunk
	[C]: in function 'require'
	__boblogistics__/data-updates.lua:10: in main chunk
   3.320 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology aluminium-processing does not exist.
   3.320 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__boblogistics__/prototypes/technology-updates.lua:169: in main chunk
	[C]: in function 'require'
	__boblogistics__/data-updates.lua:10: in main chunk
   3.320 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology titanium-processing does not exist.
   3.320 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__boblogistics__/prototypes/technology-updates.lua:188: in main chunk
	[C]: in function 'require'
	__boblogistics__/data-updates.lua:10: in main chunk
   3.320 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology advanced-electronics-3 does not exist.
   3.320 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__boblogistics__/prototypes/technology-updates.lua:189: in main chunk
	[C]: in function 'require'
	__boblogistics__/data-updates.lua:10: in main chunk
   3.320 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology ceramics does not exist.
   3.450 Loading mod bobmining 0.18.1 (data-updates.lua)
   3.580 Loading mod bobmodules 0.18.3 (data-updates.lua)
   3.712 Loading mod bobpower 0.18.6 (data-updates.lua)
   3.844 Loading mod bobrevamp 0.18.5 (data-updates.lua)
   3.976 Loading mod bobtech 0.18.3 (data-updates.lua)
   3.977 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__bobtech__/prototypes/technology/technology-updates.lua:34: in main chunk
	[C]: in function 'require'
	__bobtech__/data-updates.lua:3: in main chunk
   3.977 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology alloy-processing-1 does not exist.
   3.977 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__bobtech__/prototypes/technology/technology-updates.lua:39: in main chunk
	[C]: in function 'require'
	__bobtech__/data-updates.lua:3: in main chunk
   3.977 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology battery-2 does not exist.
   3.977 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__bobtech__/prototypes/technology/technology-updates.lua:40: in main chunk
	[C]: in function 'require'
	__bobtech__/data-updates.lua:3: in main chunk
   3.977 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology ceramics does not exist.
   4.108 Loading mod bobwarfare 0.18.6 (data-updates.lua)
   4.111 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__bobwarfare__/prototypes/technology/technology-updates.lua:134: in main chunk
	[C]: in function 'require'
	__bobwarfare__/data-updates.lua:3: in main chunk
   4.111 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology nitrogen-processing does not exist.
   4.242 Loading mod bobvehicleequipment 0.18.1 (data-updates.lua)
   4.244 Script @__boblibrary__/technology-functions.lua:234: stack traceback:
	__boblibrary__/technology-functions.lua:234: in function 'add_prerequisite'
	__bobvehicleequipment__/prototypes/recipe-updates.lua:57: in main chunk
	[C]: in function 'require'
	__bobvehicleequipment__/data-updates.lua:1: in main chunk
   4.244 Script @__boblibrary__/error-functions.lua:13: Prerequisite technology zinc-processing does not exist.
   4.375 Loading mod reskins-bobs 0.0.3 (data-updates.lua)
   4.618 Loading mod boblibrary 0.18.9 (data-final-fixes.lua)
   4.619 Script @__boblibrary__/technology-functions.lua:474: Running technology prerequisite cleanup...
   4.782 Loading mod reskins-library 0.0.1 (data-final-fixes.lua)
   4.940 Loading mod bobtech 0.18.3 (data-final-fixes.lua)
   5.099 Loading mod reskins-bobs 0.0.3 (data-final-fixes.lua)
   5.264 Checksum for core: 2798189957
   5.264 Checksum of base: 3555910100
   5.264 Checksum of bobinserters: 168175694
   5.264 Checksum of boblibrary: 3663059181
   5.264 Checksum of reskins-library: 326084899
   5.264 Checksum of bobassembly: 3686392896
   5.264 Checksum of bobequipment: 2171673949
   5.264 Checksum of bobgreenhouse: 3828947702
   5.264 Checksum of boblogistics: 1408459658
   5.264 Checksum of bobmining: 1434125394
   5.264 Checksum of bobmodules: 2867155780
   5.264 Checksum of bobpower: 988467601
   5.264 Checksum of bobrevamp: 394740161
   5.264 Checksum of bobtech: 279840677
   5.264 Checksum of bobwarfare: 3407797938
   5.264 Checksum of bobvehicleequipment: 1662738641
   5.264 Checksum of reskins-bobs: 163402949
   5.418 Error ModManager.cpp:1508: Error while loading entity prototype "pumpjack" (mining-drill): Animation speed in hr_version doesn't match normal animation speed in property tree at ROOT.mining-drill.pumpjack.animations.north.layers[0].hr_version
Modifications: Base mod › Bob's Mining mod › Artisanal Reskins: Bob's Mods › Bob's Functions Library mod › Artisanal Reskins: Core Library › Bob's Technology mod › Artisanal Reskins: Bob's Mods
   5.419 Verbose ModManager.cpp:554: Time to load mods: 5.21253
   5.551 Initial atlas bitmap size is 16384
   5.551 Created atlas bitmap 2048x560 [icon, not-compressed, mipmap, linear-minification, linear-magnification, linear-mip-level]
   5.554 Verbose TextureProcessor.cpp:1140: GPU accelerated compression test: Time 3.219 ms; YCoCg_DXT PSNR: 35.83, BC3 PSNR: 33.82
   5.554 Texture processor created (2048). GPU accelerated compression Supported: yes, Enabled: yes/yes. Test passed. YCoCgDXT PSNR: 35.83, BC3 PSNR: 33.82
   5.558 Parallel Sprite Loader initialized (threads: 5)
   5.582 Sprites loaded
   5.592 Generated mipmaps (5) for atlas [0] of size 2048x560   
   5.592 Custom mipmaps uploaded.
   5.592 Verbose AtlasSystem.cpp:910: Atlas memory size: 4.38MB; 0.00MB (virtual)
   5.592 Verbose AtlasSystem.cpp:911: Size of sprites outside of atlas: 0.00MB
   5.595 Factorio initialised
   5.600 Mods to disable:Failed to load mods: Error while loading entity prototype "pumpjack" (mining-drill): Animation speed in hr_version doesn't match normal animation speed in property tree at ROOT.mining-drill.pumpjack.animations.north.layers[0].hr_version
Modifications: Base mod › Bob's Mining mod › Artisanal Reskins: Bob's Mods › Bob's Functions Library mod › Artisanal Reskins: Core Library › Bob's Technology mod › Artisanal Reskins: Bob's Mods

Mods to be disabled:
• bobmining (0.18.1)
• reskins-bobs (0.0.3)
• boblibrary (0.18.9)
• reskins-library (0.0.1)
• bobtech (0.18.3)
• reskins-bobs (0.0.3)

```

**Comment by kirazy** [comment]
Can you tell me your mod settings, absent the reskin mod? The set of mods your using starts without issue for me. Are you normal or high graphics?

**Comment by kirazy** [comment]
Okay, got the issue to occur. Time to fix. :)

**Comment by kirazy** [comment]
Division by zero! You have the extra pumpjacks hidden, so there's only one and therefore no linear interpolation to be done, but no protection against that eventuality. :)
Commit 624ae0a closes #1.
